### PR TITLE
ath79: initial tl-wr1043nd v1 support

### PR DESCRIPTION
**Pull request by lucize**
not working: wifi, lazy port4 (when flash boot)

mentions:
network is working only with tftboot initramfs image !? (some help needed)

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

### CODE REVIEW

File: target/linux/ath79/dts/ar9132_tl_wr1043nd_v1.dts [hunk 1]
```diff
@@ -22,19 +26,18 @@
 	};
 
 	keys {
-		compatible = "gpio-keys-polled";
+		compatible = "gpio-keys";
```
Comment (blogic) [inline target/linux/ath79/dts/ar9132_tl_wr1043nd_v1.dts]: why change it to real keys ?

**Comment by blogic** [comment]
eth issue might be related to missing plldata and/or missing set_speed support inside adjust_link ndo. I have that unit and will look into it once ive done ahb wifi

File: target/linux/ath79/dts/ar9132_tl_wr1043nd_v1.dts (hunk 1 above)
Comment (lucize) [reply target/linux/ath79/dts/ar9132_tl_wr1043nd_v1.dts]: Saw somewhere that this is deprecated, they do work tho

**Comment by lucize** [comment]
The strange thing is that I see network traffic with tcpdump on every vlan interface

Tried with qca,ar7100-eth and when booting from flash is the same

**Comment by lucize** [comment]
I was exactly right now trying to add adjust_link without seeing your staging, I'll report the outcome
Thanks!

**Comment by lucize** [comment]
@blogic  the build is crashing with

```
drivers/net/phy/rtl8366_smi.c: In function 'rtl8366_reset':
drivers/net/phy/rtl8366_smi.c:322:3: error: too many arguments to function 'smi->hw_reset'
   smi->hw_reset(smi, true);
   ^~~
drivers/net/phy/rtl8366_smi.c:324:3: error: too many arguments to function 'smi->hw_reset'
   smi->hw_reset(smi, false);
   ^~~
  AR      drivers/usb/common/built-in.o
drivers/net/phy/rtl8366_smi.c: In function '__rtl8366_smi_init':
drivers/net/phy/rtl8366_smi.c:1303:3: error: too many arguments to function 'smi->hw_reset'
   smi->hw_reset(smi, false);
   ^~~
drivers/net/phy/rtl8366_smi.c: In function '__rtl8366_smi_cleanup':
drivers/net/phy/rtl8366_smi.c:1318:3: error: too many arguments to function 'smi->hw_reset'
   smi->hw_reset(smi, true);
```

```
drivers/net/phy/rtl8366_smi.c: In function 'rtl8366_smi_reset':
drivers/net/phy/rtl8366_smi.c:1431:3: error: implicit declaration of function 'reset_control_assert'; did you mean 'set_c0_brcm_reset'? [-Werror=implicit-function-declaration]
   reset_control_assert(smi->reset);
   ^~~~~~~~~~~~~~~~~~~~
   set_c0_brcm_reset
drivers/net/phy/rtl8366_smi.c:1431:29: error: 'struct rtl8366_smi' has no member named 'reset'; did you mean 'hw_reset'?
   reset_control_assert(smi->reset);
                             ^~~~~
                             hw_reset
```

```
drivers/net/phy/rtl8366_smi.c:1448:15: error: implicit declaration of function 'devm_reset_control_get'; did you mean 'devm_pinctrl_get'? [-Werror=implicit-function-declaration]
  smi->reset = devm_reset_control_get(&pdev->dev, "switch");
               ^~~~~~~~~~~~~~~~~~~~~~
               devm_pinctrl_get
drivers/net/phy/rtl8366_smi.c:1449:19: error: 'struct rtl8366_smi' has no member named 'reset'; did you mean 'hw_reset'?
  if (!IS_ERR(smi->reset))
                   ^~~~~
                   hw_reset
drivers/net/phy/rtl8366_smi.c:1450:17: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   smi->hw_reset = rtl8366_smi_reset;
                 ^
drivers/net/phy/rtl8366_smi.c: In function 'rtl8366_smi_probe_plat':
drivers/net/phy/rtl8366_smi.c:1472:16: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  smi->hw_reset = pdata->hw_reset;
                ^
  CC      drivers/usb/phy/of.o
cc1: some warnings being treated as errors
```

**Comment by lucize** [comment]
without the switch reset patch, lan and wifi is working
I did reset my tree so I don't know if I should encountered the compile problems

**Comment by blogic** [comment]
the reset patch was missing a file, its added now. I'll push my staging tree to master once the wifi patch got an ack

**Comment by lucize** [comment]
@blogic in 1043nd-v1 the wifi was working, on 841n-v9 I got data bus error and then panic, but maybe it's not the same address
I'll test on 1043nd-v3

**Pull request closed without merging**